display-test-app fails to start: esbuild can't transform `flatbush` destructuring for the configured target

### Summary
 After updating to current `master` (and running `rush install`), `display-test-app` fails to start. Vite's dependency pre-bundling (`optimizeDeps`) throws an esbuild error while processing `flatbush@4.4.0`:
 
 ```
 ✘ [ERROR] Transforming destructuring to the configured target environment
 ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides) is not supported yet
 
     flatbush@4.4.0/index.js:26:14:
       26 │ const [magic, versionAndType] = new Uint8Array(data, byteOffset + 0, 2);
 ```
 
 ### Root cause
 PR **#9398 — "Security: remediate GHSA-gv7w-rqvm-qjhr (esbuild <0.28.1)"** (commit `7a735d806b`) added a pnpm override forcing **esbuild → `~0.28.1`**. esbuild 0.28.x errors (rather than silently allowing) on array destructuring of typed arrays when targeting the app's browserslist (`chrome87`/`es2020`/...).
 
 `flatbush@4.4.0` — a long-standing dependency of `extensions/map-layers-formats` (since #6524, April 2024) — contains exactly that pattern (`const [magic, versionAndType] = new Uint8Array(...)`). DTA's Vite `optimizeDeps` pre-bundles it against the old browserslist target, so esbuild now hard-errors.
 
 The dependency is not new; the esbuild bump in **#9398** is what surfaced the incompatibility. This breaks DTA startup for everyone on `master` after #9398.
 
 ### Reproduction
 1. Check out current `master` (at or after #9398 / `7a735d806b`).
 2. `rush install`
 3. `cd test-apps/display-test-app && npm run start`
 4. Observe the esbuild transform error above.
 
 ### Possible fixes
 - Add `flatbush` to `optimizeDeps.exclude` in `test-apps/display-test-app/vite.config.mts`, **or**
 - Raise the `optimizeDeps.esbuildOptions.target` (e.g. `es2022`) so destructuring is permitted, **or**
 - Bump/patch `flatbush` to a build without typed-array destructuring.
 
 ### Notes
 - Caused by the toolchain (esbuild), **not** application source changes.
 - Unrelated to npm/Node version (repo uses Rush + pnpm; esbuild is pinned via pnpm override).

--- a/test-apps/display-test-app/vite.config.mts
+++ b/test-apps/display-test-app/vite.config.mts
@@ -179,6 +179,7 @@ export default defineConfig(() => {
         "@itwin/core-common", //prevents rpc errors
       ],
       esbuildOptions: {
+        target: "es2022", // allow modern syntax (e.g. typed-array destructuring in flatbush) during dep pre-bundling
         plugins: [
           {
             name: "externalize-electron",

